primer.md

```diff
@@ -235,6 +236,54 @@ general, this is true for all CloudEvents attributes as well.
 This section provides additional background and design points related to some of
 the CloudEvent attributes.
 
+### String Values
+
+Some of the CloudEvent attributes are defined to be of type `String`, and
+all data types used by CloudEvent attribute are required to have a
+string-encoding. This makes for an easy serialization to even the most
+basic formats and transports that might only have a string representation of
+attributes.
+
+Even with that model, it may still be possible to use certain
+characters that could be problematic in some cases. The CloudEvents
+specification does not attempt to further restrict the valid characters used
+for attribute values so as to allow for the specification to be used in
+situations where those potentially problematic characters are desired.
+
+However, this should not be interpreted as encouragement for producers to
+push the boundaries of what characters should be used just for the sake of
+edge-testing implementations. CloudEvents is striving to help with
+interoperability while still being flexible enough to be used in unique
+situations. So, CloudEvent producers are encouraged to only use values that
+have the best chance of being properly processed by all possible formats
+and transports that their CloudEvents might be used with.
+
+### URI-reference Values
+
+Attributes of type URI-reference have quite a large set of valid values that
+can be used. For example, it is possible that the `source` attribute could be
+as short as `home` or could be more meaningful, such as
+`http://home.example.com`. The CloudEvents specification allows for a wide
+range of values to allow for it to be used in as any use cases as possible,
+without undue burden on implementations.
+
+As an example, while `home` as a value for `source` would not be a wise
+choice when the CloudEvent is meant to be used in a very public environment,
+it may make perfect sense in a closed system where a full reverse-DNS name
+might be unnecessary. Or perhaps there is a severe size limitation on
+the size of the CloudEvent and shorter URI-references are needed.
+
+These are decisions that CloudEvent producers will need to decide for
+themselves based on their needs and expected usage of their events.
+
+Additionally, the CloudEvents specification does not define a `baseURI`
+to be used along with the URI-reference type attributes. This is because,
+unless otherwise stated, the URI-references are not meant to be normalized
+or used for any other purpose other than comparing its string value against
+some other URI-reference string value for equality. The use of URI-reference,
+instead of just a String, was done to encourage the use of meaningful values
```

Comment:
Referring to a complex specification such as [RFC3986](https://tools.ietf.org/html/rfc3986) just to encourage people to chose meaningful values while ignoring most of the specification, seems a bit strange.

Reply:
maybe but I think there is something to be said for encouraging some level of consistency, but still allowing a lot of freedom.  Kind of like a best-practice.  But, having said that, this is just my memory of our previous chats and if people think I'm remembering incorrectly, or they just don't want to even mention this aspect of it, I'd be ok with removing this sentence if that would help